docs/library/rp2.rst

```diff
@@ -71,6 +71,90 @@ For running PIO programs, see :class:`rp2.StateMachine`.
     This exception is raised from `asm_pio()` or `asm_pio_encode()` if there is
     an error assembling a PIO program.
 
+PIO Assembly Language instructions
+-----------------------------------
+
+Each state machine has a set of instructions, which allow for a wide range of
+behaviourial or location changes to the data. They are written in assembly
+language which is later used in a MicroPython program to perform specific
+tasks. These instructions are -
+
+jmp(cond, label=None)
+   This can take two forms: 
+   jmp(label)
+       label : label to jump to unconditionlly
+
+    jmp(cond,label)
+        cond: not_x, x_dec, not_y, y_dec, x_not_y, pin, not_osre
+        label: label to jump to if condition is true   
+    
+    see sec 3.4.2 of RPi docs for details
+
+wait(polarity, src, index)
+     polarity: 0-1
+     src: gpio, pin, irq
+     index: 0-31
+     wait for high/low on pin 0-31 or irq 0-31
+     see sec 3.4.3 of RPi docs for details
+
+in(src, data)
+    src : pin, x, y, isr, osr(all 0-31)
+    data : bit count(0-31)
+    see sec 3.4.4 of RPi docs for details
+
+out(dest,data)
+    dest : pin, x, y, pindirs, ac, isr, osr(all 0-31)
+    data : bit count(0-31)
+    see sec 3.4.5 of RPi docs for details
+
+push()
+    This instruction takes the following forms:
+    push()
+    push(block)
+    push(noblock)
+    push(iffull)
+    push(iffull, block)
+    push(iffull, noblock)
+    see sec 3.4.6 of RPi docs for details
+
+pull()
+    This instruction can take the following forms:
+    pull()
+    pull(block)
+    pull(noblock)
+    pull(ifempty)
+    pull(ifempty, block)
+    pull(ifempty, noblock) 
+    see sec 3.4.7 of RPi docs for details
+
+mov(dest, src)
+    dest : pin, x, y, isr, osr, pc (all 0-31)
+    src : pin, x, y, isr, osr (all 0-31)
+    see sec 3.4.8 of RPi docs for details
+
+set(dest, data)
+    dest : x, y, pins, pindirs
+    data : value (0-31)
+    see sec 3.4.10 of RPi docs for details
+
+irq(mod, index = None)
+    These are the following forms of this instruction(index is an integer value 
+    from 0-7)
+    
+    These can be summarised as two forms:
+    * form 1: irq(index)
+    * form 2: irq(mode,index)
+      where
+      * index can be an integer or rel(integer)
+      * mode can be: block or clear or block | clear
+      
+    Allowed IRQ numbers are 0-7 (0-3 are visible from to the processor, 4-7 are 
+    internal to the state machines).          
+   see sec 3.4.9 of RPi docs for details
+
+nop(self)
+    see sec 3.3.7 of RPi docs for details
+
```

Comment:
Also need to add here the delay and side-set modifiers.

Reply:
This still needs to be done (describing delay and side-set).  Also list the 4 directives in this reference.